server/controllers/api/video-channel-sync.ts

```diff
@@ -0,0 +1,103 @@
+import { auditLoggerFactory, getAuditIdFromRes, VideoChannelSyncAuditView } from '@server/helpers/audit-logger'
+import { logger } from '@server/helpers/logger'
+import { sequelizeTypescript } from '@server/initializers/database'
+import { JobQueue } from '@server/lib/job-queue'
+import {
+  asyncMiddleware,
+  asyncRetryTransactionMiddleware,
+  authenticate,
+  ensureCanManageChannel as ensureCanManageSyncedChannel,
+  ensureCanUpload,
+  ensureSyncExists,
+  ensureSyncIsEnabled,
+  videoChannelSyncValidator
+} from '@server/middlewares'
+import { VideoChannelSyncModel } from '@server/models/video/video-channel-sync'
+import { HttpStatusCode, VideoChannelSyncState } from '@shared/models'
+import express from 'express'
+
+const videoChannelSyncRouter = express.Router()
+const auditLogger = auditLoggerFactory('channels')
+
+videoChannelSyncRouter.post('/',
+  authenticate,
+  ensureSyncIsEnabled,
+  asyncMiddleware(videoChannelSyncValidator),
+  ensureCanManageSyncedChannel,
+  asyncRetryTransactionMiddleware(createVideoChannelSync)
+)
+
+videoChannelSyncRouter.post('/:id/sync',
+  authenticate,
+  ensureSyncIsEnabled,
+  asyncMiddleware(ensureSyncExists),
+  ensureCanManageSyncedChannel,
+  asyncMiddleware(ensureCanUpload),
+  syncChannel
+)
+
+videoChannelSyncRouter.delete('/:id',
+  authenticate,
+  asyncMiddleware(ensureSyncExists),
+  ensureCanManageSyncedChannel,
+  asyncRetryTransactionMiddleware(removeVideoChannelSync)
+)
+
+export { videoChannelSyncRouter }
+
+// ---------------------------------------------------------------------------
+
+async function createVideoChannelSync (req: express.Request, res: express.Response) {
+  const syncCreated = new VideoChannelSyncModel({
+    externalChannelUrl: req.body.externalChannelUrl,
+    videoChannelId: req.body.videoChannelId,
+    state: VideoChannelSyncState.WAITING_FIRST_RUN
+  })
+  await sequelizeTypescript.transaction(async t => {
+    await syncCreated.save({ transaction: t })
+    await syncCreated.reload({ transaction: t })
```

Comment:
Why do you need this reload?

Reply:
Otherwise I don't retrieve the `syncCreated.VideoChannel` property (even with the `@DefaultScope` added in the model).

If you have any tip to avoid the reload, I would appreciate!

Reply:
In fact you should check video channel existence in middleware and populate `res.locals.videoChannel`. Then you can just do `syncCreated.VideoChannel = res.locals.videoChannel`